setup.ts

```diff
@@ -0,0 +1,379 @@
+  let DB_URL = process.env.MONGO_DB_URL;
+
+  try {
+    let isConnected = false,
+      url = "";
+    while (!isConnected) {
+      url = await askForMongoDBUrl();
+      isConnected = await checkConnection(url);
+    }
+
+    DB_URL = url;
+    const config = dotenv.parse(fs.readFileSync(".env"));
+    config.MONGO_DB_URL = DB_URL;
+    process.env.MONGO_DB_URL = DB_URL;
+    fs.writeFileSync(".env", "");
+    for (const key in config) {
+      fs.appendFileSync(".env", `${key}=${config[key]}\n`);
+    }
+  } catch (err) {
+    console.error(err);
+    abort();
+  }
+}
+
+//Get recaptcha details
+async function recaptcha(): Promise<void> {
+  console.log(
+    "\nPlease visit this URL to set up reCAPTCHA:\n\nhttps://www.google.com/recaptcha/admin/create"
+  );
+  console.log(
+    '\nSelect reCAPTCHA v2 and the "I`m not a robot" checkbox option'
+  );
+  console.log(
+    '\nAdd "localhost" in domains and accept the terms, then press submit'
+  );
+
+  const { recaptchaSecretKey } = await inquirer.prompt([
+    {
+      type: "input",
+      name: "recaptchaSecretKey",
+      message: "Enter your reCAPTCHA secret site key:",
+      validate: async (input: string): Promise<boolean | string> => {
+        if (validateRecaptcha(input)) {
+          return true;
+        }
+        return "Invalid reCAPTCHA secret key. Please try again.";
+      },
+    },
+  ]);
+  const config = dotenv.parse(fs.readFileSync(".env"));
+  config.RECAPTCHA_SECRET_KEY = recaptchaSecretKey;
+  fs.writeFileSync(".env", "");
+  for (const key in config) {
+    fs.appendFileSync(".env", `${key}=${config[key]}\n`);
+  }
+}
+
+function isValidEmail(email: string): boolean {
+  const pattern = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+  const match = email.match(pattern);
+  return match !== null && match[0] === email;
+}
+
+function validateRecaptcha(string: string): boolean {
+  const pattern = /^[a-zA-Z0-9_-]{40}$/;
+  return pattern.test(string);
+}
+
+function abort(): void {
+  console.log("\nSetup process aborted. 🫠");
+  process.exit(1);
+}
+
+//Get mail username and password
+async function twoFactorAuth(): Promise<void> {
+  console.log("\nIMPORTANT");
+  console.log(
+    "\nEnsure that you have Two-Factor Authentication set up on your Google Account."
+  );
+  console.log("\nVisit Url: https://myaccount.google.com");
+  console.log(
+    "\nSelect Security and under Signing in to Google section select App Passwords."
+  );
+  console.log(
+    "\nClick on Select app section and choose Other(Custom name), enter talawa as the custom name and press Generate button."
+  );
+
+  const { email, password } = await inquirer.prompt([
+    {
+      type: "input",
+      name: "email",
+      message: "Enter your email:",
+      validate: (input: string) =>
+        isValidEmail(input) || "Invalid email. Please try again.",
+    },
+    {
+      type: "password",
+      name: "password",
+      message: "Enter the generated password:",
+      mask: "*",
```

Comment:
this is just a setup script, so i don't think masking it is necessary

Reply:
Removed masking